hudi-client/src/main/java/org/apache/hudi/metadata/FSBackedTableMetadataWriter.java

```diff
@@ -74,84 +60,109 @@
 import org.apache.hudi.exception.HoodieMetadataException;
 import org.apache.hudi.metrics.DistributedRegistry;
 import org.apache.hudi.table.HoodieTable;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
 import org.apache.log4j.LogManager;
 import org.apache.log4j.Logger;
 import org.apache.spark.api.java.JavaPairRDD;
 import org.apache.spark.api.java.JavaRDD;
 import org.apache.spark.api.java.JavaSparkContext;
 
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
 import scala.Tuple2;
 
+import static org.apache.hudi.metadata.HoodieTableMetadata.METADATA_TABLE_NAME_SUFFIX;
+import static org.apache.hudi.metadata.HoodieTableMetadata.NON_PARTITIONED_NAME;
+import static org.apache.hudi.metadata.HoodieTableMetadata.SOLO_COMMIT_TIMESTAMP;
+
 /**
  * Writer for Metadata Table.
  *
  * Partition and file listing are saved within an internal MOR table called Metadata Table. This table is created
  * by listing files and partitions (first time) and kept in sync using the instants on the main dataset.
  */
-public class HoodieMetadataWriter extends HoodieMetadataReader implements Serializable {
-  private static final Logger LOG = LogManager.getLogger(HoodieMetadataWriter.class);
+public class FSBackedTableMetadataWriter implements HoodieTableMetadataWriter {
```

Comment:
Is FSBased the appropriate name here? FSBased seems to imply that the metadata is being saved directly on filesystem. But technically we are saving the metadata in a hoodie table.

Reply:
Good point. Call it - `HoodieBackedTableMetadataWriter`?  wdyt @prashantwason

Reply:
Yep, that sounds fine.